libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,430 @@
+import static au.com.dius.pact.consumer.dsl.LambdaDsl.newJsonBody;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+import static uk.gov.di.ipv.core.library.enums.EvcsVCState.PENDING_RETURN;
+import static uk.gov.di.ipv.core.library.enums.EvcsVcProvenance.ONLINE;
+
+@ExtendWith(PactConsumerTestExt.class)
+@ExtendWith(MockitoExtension.class)
+@PactTestFor(providerName = "EvcsProvider")
+@MockServerConfig(hostInterface = "localhost")
+class ContractTest {
+    private static final String EVCS_API_KEY =
+            "L2BGccX59Ea9PMJ3ipu9t7r99ykD2Tlh1KYpdjdg"; // pragma: allowlist secret
+    private static final String TEST_EVCS_ACCESS_TOKEN = "TEST_EVCS_ACCESS_TOKEN";
+    private static final String TEST_USER_ID = "9bd7f130-4238-4532-83cd-01cb29584834";
+    private static final String INVALID_USER_ID = "invalid-user-id";
+    private static final List<EvcsVCState> VC_STATES_FOR_QUERY = List.of(PENDING_RETURN);
+
+    private static final List<EvcsCreateUserVCsDto> EVCS_CREATE_USER_VCS_DTO =
+            List.of(
+                    new EvcsCreateUserVCsDto(
+                            VcFixtures.VC_ADDRESS.getVcString(),
+                            EvcsVCState.CURRENT,
+                            Map.of(
+                                    "reason", "testing",
+                                    "txmaEventId", "txma-event-id-2",
+                                    "timestampMs", "1714478033959"),
+                            ONLINE));
+
+    private static final List<EvcsUpdateUserVCsDto> EVCS_UPDATE_USER_VCS_DTO =
+            List.of(
+                    new EvcsUpdateUserVCsDto(
+                            "VC_Signature1",
+                            EvcsVCState.HISTORIC,
+                            Map.of(
+                                    "reason", "testing",
+                                    "txmaEventId", "txma-event-id-2",
+                                    "timestampMs", "1714478033959")));
+
+    @Mock ConfigService mockConfigService;
+
+    @BeforeEach
+    void setUp(MockServer mockServer) {
+        when(mockConfigService.getParameter(ConfigurationVariable.EVCS_APPLICATION_URL))
+                .thenReturn("http://localhost:" + mockServer.getPort());
+        lenient()
+                .when(mockConfigService.getSecret(ConfigurationVariable.EVCS_API_KEY))
+                .thenReturn(EVCS_API_KEY);
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "EvcsConsumer")
+    public RequestResponsePact validRetrieveVcRequestReturnsUsersVcWith200(
+            PactDslWithProvider builder) {
+        return builder.given("EVCS client exist")
+                .given("VCS exist")
+                .given("EVC has users VC")
```

Comment:
I'm not quite sure what all these givens are supposed to mean - I think they should be clear about what state is expected to be set up - i.e. all the test access tokens, user ids etc.

How about something like:
```
.given("test-api-key is a valid API key")
.given("test-acess-token is a valid access token")
.given("test-user-id has one PENDING_RETURN VC")
...
```

This gives you a pattern for setting up other tests, where one of those conditions may not be true!